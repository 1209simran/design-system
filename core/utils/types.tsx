export type BaseProps = {
  className?: string,
  'data-test'?: string,
};

const baseProps = [
  'className',
  'data-test'
];

export const extractBaseProps = (props: Record<string, any>) => {
  const basePropsObj = baseProps.reduce((acc, curr) => {
    return (
      props[curr] ? { ...acc, [curr]: props[curr] } : { ...acc }
    );
  }, {});

  return basePropsObj;
};
