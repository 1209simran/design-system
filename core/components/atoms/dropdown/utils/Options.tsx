export let dropdownOptions: any[] = [];
export const group1: any[] = [];
const group2: any[] = [];

for (let i = 1; i <= 40; i++) {
  group1.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

for (let i = 41; i <= 50; i++) {
  group2.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

dropdownOptions.push({
  group: true,
  label: 'Group 1',
  items: group1,
});

dropdownOptions.push({
  group: true,
  label: 'Group 2',
  items: group2,
});

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = group1.slice(0, 10);

export const storyWrapOptions = [
  {
    label: 'Design System Dropdown',
    value: 'Design System Dropdown'
  },
  {
    label: 'UI Kit Dropdown',
    value: 'UI Kit Dropdown',
  },
  {
    label: 'Innovaccer Analytics',
    value: 'Innovaccer Analytics'
  }
];

export const subInfoItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    icon: 'events'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    icon: 'events',
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    icon: 'events',
  },
];
