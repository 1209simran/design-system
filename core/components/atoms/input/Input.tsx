import * as React from 'react';
import classNames from 'classnames';
// import { Size } from '@/components/atoms/button';
import Text from '@/components/atoms/text';
import Popover from '@/components/molecules/popover';
import Icon, { IconProps } from '@/components/atoms/icon';

export type InputType = 'text' | 'password' | 'number' | 'email' | 'tel' | 'url';
export type AutoComplete = 'on' | 'off';
export type Size = 'tiny' | 'regular' | 'large';

export interface InputProps {
  /**
   * Name of the `Input`
   */
  name?: string;
  /**
   * Type of text inside `Input`
   */
  type?: InputType;
  /**
   * Value of the `Input`
   */
  value?: string;
  /**
   * Adds default value to `Input`
   */
  defaultValue?: string;
  /**
   * Text to display when input is empty
   */
  placeholder?: string;
  /**
   * Specifies whether `input field should have autocomplete enabled
   */
  autocomplete?: AutoComplete;
  /**
   * Size of the `Input`
   * @default "regular"
   */
  size?: Size;
  /**
   * Material icon name
   */
  icon?: string;
  /**
   * Label to be displayed inside `Input`
   */
  inlineLabel?: string;
  /**
   * Adds loader inside input when waiting for an action to complete
   */
  loading?: boolean;
  /**
   * Disables the `Input`, making it unable to type
   *
   * **set to `true` if onChange is not provided**
   */
  disabled?: boolean;
  /**
   * Shows the user that this field id required
   */
  required?: boolean;
  /**
   * Shows error state in case of failed validation
   */
  error?: boolean;
  /**
   * Text to be rendered in info `Popover`
   * @default true
   */
  info?: string;
  /**
   * Adds autoFocus
   */
  autoFocus?: boolean;
  /**
   * Callback function when user clicks the clear button
   */
  onClear?: (e: React.MouseEvent<HTMLElement>) => void;
  /**
   * Callback function when `Input` text changes
   */
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  /**
   * Handler to be called when `Input` is clicked
   */
  onClick?: (e: React.MouseEvent<HTMLInputElement>) => void;
  /**
   * Handler to be called when `Input` loses focus
   */
  onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
  /**
   * Handler to be called when `Input` gets focus
   */
  onFocus?: (e: React.FocusEvent<HTMLInputElement>) => void;
  /**
   * Custom Icon Component to be passed to Input to replace Clear Icon in the right
   */
  actionIcon?: React.ReactElement<IconProps>;
}

const sizeMapping = {
  tiny: 12,
  regular: 16,
  large: 20,
};

export const Input = React.forwardRef<HTMLInputElement, InputProps>((props, ref) => {
  const {
    size = 'regular',
    type = 'text',
    disabled,
    defaultValue,
    name,
    placeholder,
    value,
    icon,
    inlineLabel,
    required,
    error,
    info,
    autocomplete,
    autoFocus,
    onChange,
    onClick,
    onClear,
    onBlur,
    onFocus,
    actionIcon
  } = props;

  const classes = classNames({
    ['Input']: true,
    [`Input--${size}`]: size,
    ['Input--disabled']: disabled,
    ['Input--error']: error
  });

  const inputClass = classNames({
    ['Input-input']: true,
    [`Input-input--${size}`]: size,
  });

  const leftIconClass = classNames({
    ['Input-icon']: true,
    ['Input-icon--left']: true,
    ['Input-icon--disabled']: !value
  });

  const rightIconClass = classNames({
    ['Input-icon']: true,
    ['Input-icon--right']: true
  });

  const trigger = <div className={rightIconClass}><Icon name={'info'} size={sizeMapping[size]} /></div>;

  const popoverStyle = {
    padding: 'var(--spacing) var(--spacing-2)',
    maxWidth: 'var(--spacing-9)',
    overflow: 'hidden'
  };

  return (
    <div className={classes}>
      {inlineLabel && (
        <div className="Input-inlineLabel">
          <Text appearance="subtle">{inlineLabel}</Text>
        </div>
      )}
      {size !== 'tiny' && icon && (
        <div className={leftIconClass}>
          <Icon
            name={icon}
            size={sizeMapping[size]}
          />
        </div>
      )}
      <input
        ref={ref}
        name={name}
        type={type}
        defaultValue={defaultValue}
        placeholder={placeholder}
        className={inputClass}
        value={value}
        autoComplete={autocomplete}
        required={required}
        disabled={disabled}
        onChange={onChange}
        onBlur={onBlur}
        onClick={onClick}
        onFocus={onFocus}
        autoFocus={autoFocus}
      />
      {(!value && !disabled) || (value && disabled) || (defaultValue && disabled)
        ? (
          info && (
            <Popover
              style={popoverStyle}
              position="top"
              on={'hover'}
              trigger={trigger}
              dark={true}
            >
              {info}
            </Popover>
          )
        ) : (
          actionIcon
            ? (
              actionIcon
            ) : (
              (onClear && value && !disabled) && (
                <div className={rightIconClass} onClick={e => onClear(e)}>
                  <Icon name={'close'} size={sizeMapping[size]} />
                </div>
              )
            )
        )
      }
    </div>
  );
});

Input.displayName = 'Input';

export default Input;
