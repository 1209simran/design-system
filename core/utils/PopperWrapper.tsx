import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Manager, Reference, Popper } from 'react-popper';

type PositionType =
  | 'auto-start'
  | 'auto'
  | 'auto-end'
  | 'top-start'
  | 'top'
  | 'top-end'
  | 'right-start'
  | 'right'
  | 'right-end'
  | 'bottom-end'
  | 'bottom'
  | 'bottom-start'
  | 'left-end'
  | 'left'
  | 'left-start'
  | undefined;

type actionType = 'click' | 'hover';

enum Offsets {
  Small = '2px',
  Medium = '4px',
  Large = '8px'
}

interface IProps {
  trigger: React.ReactElement<any>;
  placement: PositionType;
  children: React.ReactElement<any>;
  style: React.CSSProperties;
  appendToBody: boolean;
  on?: actionType;
  offset: keyof typeof Offsets;
  closeOnBackdropClick?: boolean;
  hoverable?: boolean;
}

interface IState {
  open: boolean;
  mouseLeaveDelay: number;
  mouseEnterDelay: number;
}

class PopperWrapper extends React.Component<IProps, IState> {
  private triggerRef: React.RefObject<HTMLElement>;
  private popupRef: React.RefObject<HTMLDivElement>;
  private _timer?: number;

  constructor(props: IProps) {
    super(props);

    this.state = {
      open: false,
      mouseLeaveDelay: 50,
      mouseEnterDelay: 0
    };

    this.triggerRef = React.createRef();
    this.popupRef = React.createRef();
  }

  public componentWillUnmount() {
    clearTimeout(this._timer);
    document.removeEventListener('mousedown', this.doesNodeContainClick);
  }

  public handleMouseLeave = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: false });
        if (this.props.children.props.onMouseLeave) {
          this.props.children.props.onMouseLeave(event);
        }
      }, this.state.mouseLeaveDelay);
    } else {
      this.setState({ open: false });
      if (this.props.children.props.onMouseLeave) {
        this.props.children.props.onMouseLeave(event);
      }
    }
  }

  public handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: true });
        if (this.props.children.props.onMouseEnter) {
          this.props.children.props.onMouseEnter(event);
        }
      }, this.state.mouseEnterDelay);
    } else {
      this.setState({ open: true });
      if (this.props.children.props.onMouseEnter) {
        this.props.children.props.onMouseEnter(event);
      }
    }
  }

  public togglePopper = () => {
    this.setState({ open: !this.state.open });
  }

  public doesNodeContainClick = (event: Event) => {
    if (
      !(
        this.findDOMNode(this.popupRef).contains(event.target as HTMLElement) ||
        this.findDOMNode(this.triggerRef).contains(event.target as HTMLElement)
      )
    ) {
      this.togglePopper();
    }
  }

  public componentDidMount() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public componentDidUpdate() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    } else if (on === 'click' && !open && closeOnBackdropClick) {
      document.removeEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public getTriggerElement(trigger: React.ReactElement<any>, ref: React.Ref<any>, on: actionType) {
    const options =
      on === 'hover'
        ? {
          ref,
          onMouseEnter: this.handleMouseEnter,
          onMouseLeave: this.handleMouseLeave
        }
        : {
          ref,
          onClick: this.togglePopper
        };

    const element = React.cloneElement(
      (
        <span className="d-inline-block">
          {trigger}
        </span>
      ),
      options
      );

    return element;
  }

  public getChildrenElement(
    children: React.ReactElement<any>,
    ref: React.Ref<any>,
    placement: string,
    style: React.CSSProperties
  ) {
    const options = this.props.on === 'hover' ? {
      ref,
      style,
      onMouseEnter: this.handleMouseEnter,
      onMouseLeave: this.handleMouseLeave,
      'data-placement': placement
    } : {
      ref,
      style,
      'data-placement': placement
    };

    const element = React.cloneElement(children, options);
    return element;
  }

  public render() {
    const { trigger, children, placement, appendToBody, on = 'click', offset } = this.props;
    const { open } = this.state;

    return (
      <Manager>
        <Reference innerRef={this.triggerRef}>{({ ref }) => this.getTriggerElement(trigger, ref, on)}</Reference>
        {open &&
          appendToBody &&
          ReactDOM.createPortal(
            (
              /* tslint:disable:no-shadowed-variable */
              <Popper placement={placement} innerRef={this.popupRef}>
                {({ ref, style, placement }) => {
                  const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
                  return this.getChildrenElement(children, ref, placement, newStyle);
                }}
              </Popper>
            ),
            document.body
          )}
        {open && !appendToBody && (
          <Popper placement={placement} innerRef={this.popupRef}>
            {({ ref, style, placement }) => {
              const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
              return this.getChildrenElement(children, ref, placement, newStyle);
            }}
          </Popper>
        )}
      </Manager>
    );
  }

  private getUpdatedStyle = (
    oldStyle: React.CSSProperties,
    placement: PositionType,
    offset: keyof typeof Offsets = 'Medium'
  ) => {
    const { style = {} } = this.props;
    const newStyle = { ...style, ...oldStyle };
    const position = placement ? placement.split('-')[0] : placement;
    switch (position) {
      case 'top':
        newStyle.marginBottom = Offsets[offset];
        break;

      case 'bottom':
        newStyle.marginTop = Offsets[offset];
        break;

      case 'left':
        newStyle.marginRight = Offsets[offset];
        break;

      case 'right':
        newStyle.marginLeft = Offsets[offset];
        break;
    }
    return newStyle;
  }

  private findDOMNode = (ref: React.RefObject<HTMLElement>) => {
    return ReactDOM.findDOMNode(ref.current!) as HTMLElement;
  }
}

export default PopperWrapper;
