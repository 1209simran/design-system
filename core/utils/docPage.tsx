// @ts-nocheck

import * as React from 'react';
import { Title, Props, Description, Primary, Story, PropsProps } from '@storybook/addon-docs/blocks';

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
}

export const docPage = (props: DocPageProps) => {
  const {
    title,
    description,
    props,
    stories
  } = props;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <Description>
        {description}
      </Description>
      {/* <Primary /> */}
      {/* <Story id="switch-state--alert"  /> */}
      <Props {...props} />
    </>
  );
};

export default docPage;
