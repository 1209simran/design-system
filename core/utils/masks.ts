import { Mask } from '@/utils/types';
import { DatePickerProps } from "@/index.type";

type DateFormat = DatePickerProps['inputFormat'];
export const date: Record<DateFormat, Mask> = {
  'dd/mm/yyyy': [/[0123]/, /\d/, '/', /\[01]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'mm/dd/yyyy': [/[01]/, /\d/, '/', /[0123]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'yyyy/mm/dd': [/\d/, /\d/, /\d/, /\d/, '/', /[01]/, /\d/, '/', /[0123]/, /\d/],
  'dd-mm-yyyy': [/[0123]/, /\d/, '-', /[01]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'mm-dd-yyyy': [/[01]/, /\d/, '-', /[0123]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'yyyy-mm-dd': [/\d/, /\d/, /\d/, /\d/, '-', /[01]/, /\d/, '-', /[0123]/, /\d/],
};
