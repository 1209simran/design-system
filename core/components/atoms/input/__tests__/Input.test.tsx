import * as React from 'react';
import { shallow } from 'enzyme';
import { TestHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/TestHelper';
import Input, { IInputProps as IProps } from '../Input';

const BooleanValue = [true, false];
const StringValue = ['events'];
const NameValue = 'name';
const NumberValue = [10];
const InputType = ['text', 'password', 'number'];
const Size = ['tiny', 'regular', 'large'];
const FunctionValue = [jest.fn()];

const Mapper: Record<string, any> = {
  name: valueHelper(NameValue, { required: true }),
  size: valueHelper(Size, { iterate: true }),
  label: valueHelper(StringValue, { iterate: true }),
  inlineLabel: valueHelper(StringValue, { iterate: true }),
  type: valueHelper(InputType, { iterate: true }),
  value: valueHelper([...StringValue, ...NumberValue], { iterate: true }),
  loading: valueHelper(BooleanValue, { iterate: true }),
  icon: valueHelper(StringValue, { iterate: true }),
  placeholder: valueHelper(StringValue, { iterate: true }),
  disabled: valueHelper(BooleanValue, { iterate: true }),
  required: valueHelper(BooleanValue, { iterate: true }),
  error: valueHelper(BooleanValue, { iterate: true }),
  caption: valueHelper(StringValue, { iterate: true }),
  clearButton: valueHelper(BooleanValue, { iterate: true }),
  info: valueHelper(StringValue, { iterate: true }),
  onClearHandler: valueHelper(FunctionValue, { iterate: true }),
  onChangeHandler: valueHelper(FunctionValue, { iterate: true }),
  onClickHandler: valueHelper(FunctionValue, { iterate: true }),
};

describe('Input component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Input
          {...attr}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  TestHelper(Mapper, testFunc);
});
