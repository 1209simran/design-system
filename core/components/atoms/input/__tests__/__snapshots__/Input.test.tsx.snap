// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
>
  <input
    className="Input-input Input-input--regular"
    name="name"
    placeholder="Placeholder"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]", icon: "events"
 1`] = `
<div
  className="Input Input--regular"
>
  <div
    className="Input-icon Input-icon--left Input-icon--disabled"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
      type="filled"
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    name="name"
    placeholder="Placeholder"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]", value: "value"
 1`] = `
<div
  className="Input Input--regular"
>
  <input
    className="Input-input Input-input--regular"
    name="name"
    placeholder="Placeholder"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]", value: "value", icon: "events"
 1`] = `
<div
  className="Input Input--regular"
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
      type="filled"
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    name="name"
    placeholder="Placeholder"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "large", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--large"
>
  <input
    className="Input-input Input-input--large"
    name="name"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "large", type: "number", inlineLabel: "USD", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--large"
>
  <div
    className="Input-inlineLabel"
  >
    <Text
      appearance="subtle"
    >
      USD
    </Text>
  </div>
  <input
    className="Input-input Input-input--large"
    name="name"
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "regular", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
>
  <input
    className="Input-input Input-input--regular"
    name="name"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "regular", type: "number", inlineLabel: "USD", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
>
  <div
    className="Input-inlineLabel"
  >
    <Text
      appearance="subtle"
    >
      USD
    </Text>
  </div>
  <input
    className="Input-input Input-input--regular"
    name="name"
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "tiny", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--tiny"
>
  <input
    className="Input-input Input-input--tiny"
    name="name"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "tiny", type: "number", inlineLabel: "USD", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--tiny"
>
  <div
    className="Input-inlineLabel"
  >
    <Text
      appearance="subtle"
    >
      USD
    </Text>
  </div>
  <input
    className="Input-input Input-input--tiny"
    name="name"
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", type: "number", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
>
  <input
    className="Input-input Input-input--regular"
    name="name"
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", type: "password", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
>
  <input
    className="Input-input Input-input--regular"
    name="name"
    type="password"
  />
</div>
`;

exports[`Input component 
  name: "name", type: "text", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
>
  <input
    className="Input-input Input-input--regular"
    name="name"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true
 1`] = `
<div
  className="Input Input--regular Input--disabled"
>
  <input
    className="Input-input Input-input--regular"
    disabled={true}
    name="name"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true, icon: "events"
 1`] = `
<div
  className="Input Input--regular Input--disabled"
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
      type="filled"
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    disabled={true}
    name="name"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true, onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular Input--disabled"
>
  <input
    className="Input-input Input-input--regular"
    disabled={true}
    name="name"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true, onChangeHandler: "[Function]", icon: "events"
 1`] = `
<div
  className="Input Input--regular Input--disabled"
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
      type="filled"
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    disabled={true}
    name="name"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", error: true, onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular Input--error"
>
  <input
    className="Input-input Input-input--regular"
    name="name"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", error: true, onChangeHandler: "[Function]", icon: "events"
 1`] = `
<div
  className="Input Input--regular Input--error"
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
      type="filled"
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    name="name"
    type="text"
    value="value"
  />
</div>
`;
